src/routes/safe/components/Transactions/TxList/TxOwners.tsx

```diff
@@ -24,92 +123,79 @@ export const TxOwners = ({
+        </StyledStep>
+      ) : (
+        <StyledStep bold state="confirmed">
+          <StepLabel icon={<TxCreationIcon />}>Created</StepLabel>
+        </StyledStep>
+      )}
+      <StyledStep bold state={isConfirmed ? 'confirmed' : 'active'}>
+        <StepLabel icon={isConfirmed ? <CheckIcon /> : <CircleIcon />}>
+          Confirmations{' '}
+          <span style={confirmationsStyle}>
+            ({`${detailedExecutionInfo.confirmations.length} of ${detailedExecutionInfo.confirmationsRequired}`})
           </span>
-          <div className="legend">
-            <Text color="primary" size="xl" strong>
-              {detailedExecutionInfo.executor ? 'Executed' : isPending ? 'Executing' : 'Execute'}
-            </Text>
-            {detailedExecutionInfo.executor && (
+        </StepLabel>
+      </StyledStep>
+      {!hideConfirmations &&
+        detailedExecutionInfo.confirmations.map(({ signer }) => (
+          <StyledStep key={signer.value} bold state="confirmed">
+            <StepLabel icon={<DotIcon />}>
+              <AddressInfo
+                address={signer.value}
+                name={signer?.name || undefined}
+                avatarUrl={signer?.logoUri || undefined}
+                shortenHash={4}
+              />
+            </StepLabel>
+          </StyledStep>
+        ))}
+      {detailedExecutionInfo.confirmations.length > 0 && (
+        <StyledStep state="confirmed">
+          <StepLabel icon={<DotIcon />} onClick={toggleHide}>
+            <span style={pointerStyle}>{hideConfirmations ? 'Show all' : 'Hide all'}</span>
+          </StepLabel>
+        </StyledStep>
+      )}
+      <StyledStep expanded bold state={isExecuted ? 'confirmed' : 'disabled'}>
+        <StepLabel icon={isExecuted ? <CheckIcon /> : <CircleIcon />}>
+          {isExecuted ? 'Executed' : isPending ? 'Executing' : 'Execution'}
+        </StepLabel>
+        {
+          // isExecuted
+          detailedExecutionInfo.executor ? (
+            <StepContent>
               <AddressInfo
                 address={detailedExecutionInfo.executor.value}
                 name={detailedExecutionInfo.executor?.name || undefined}
                 avatarUrl={detailedExecutionInfo.executor?.logoUri || undefined}
                 shortenHash={4}
               />
-            )}
-          </div>
-        </OwnerListItem>
-      ) : (
-        <OwnerListItem>
-          <span className="icon">
-            <StyledImg alt="" src={TransactionListInactive} />
-          </span>
-          <div className="legend">
-            <Text color="icon" size="xl" strong>
-              Execute ({confirmationsNeeded} more {confirmationsNeeded === 1 ? 'confirmation' : 'confirmations'} needed)
-            </Text>
-          </div>
-        </OwnerListItem>
-      )}
-    </OwnerList>
+            </StepContent>
+          ) : (
+            !isConfirmed && <StyledStepContent>Can be executed once the threshold is reached</StyledStepContent>
```

Comment:
In case the last signer wants to immediately execute we should also hide this message. That case should be covered if we extend the condition to `!isConfirmed && !isPending`